lib/src/main/java/io/ably/lib/types/ProtocolMessage.java

```diff
@@ -205,4 +214,42 @@ public JsonElement serialize(Action action, Type t, JsonSerializationContext ctx
 			return new JsonPrimitive(action.getValue());
 		}		
 	}
+
+	public static class AuthDetails {
+		public String accessToken;
+
+		private AuthDetails() { }
+		public AuthDetails(String s) { accessToken = s; }
+
+		AuthDetails readMsgpack(MessageUnpacker unpacker) throws IOException {
+			int fieldCount = unpacker.unpackMapHeader();
+			for(int i = 0; i < fieldCount; i++) {
+				String fieldName = unpacker.unpackString().intern();
+				MessageFormat fieldFormat = unpacker.getNextFormat();
+				if(fieldFormat.equals(MessageFormat.NIL)) { unpacker.unpackNil(); continue; }
+
+				if(fieldName == "accessToken") {
+					accessToken = unpacker.unpackString();
+				} else {
+					System.out.println("Unexpected field: " + fieldName);
```

Comment:
That's cut+paste from other existing code that does the same thing. I'm not sure we should even log it tbh - it's legitimate for a message to contain unknown fields when there is a protocol change, for example.